### PR TITLE
channeldb+routing+gossiper: add local updates to graph immediately

### PR DESCRIPTION
**Pull request by halseth**
Since the batch interval can potentially be long, adding local updates
to the graph could be slow. This would slow down operations like adding
our own channel update and announcements during the funding process, and
updating edge policies for local channels.

Now we instead check whether the update is remote or not, and use the
SchedulerOption to immediately add local updates to the graph.

Companion PR to #4958

### CODE REVIEW

File: batch/interface.go
Comment (halseth) [inline batch/interface.go:40]: Could turn this on its head, and make priority add the default. Then remote updates could be given a `LazyAdd` option to indicate they can be batched.

Comment (halseth) [reply batch/interface.go]: made lazy the option

**Review by cfromknecht — APPROVED** [review]
LGTM 🎉

**Review by Crypt-iQ — APPROVED** [review]
Did not test, but changes seem straightforward. One nit

**Review by Roasbeef — APPROVED** [review]
LGTM 🤽‍♂️

**Comment by Roasbeef** [comment]
There appears to be an itest failure related to private channels.

**Comment by halseth** [comment]
The flakes seems to just be the same as those seen in master, not clear whether this PR makes it worse.

I think we should wait for this to be merged until we have #4953 or similar merged.

**Comment by Roasbeef** [comment]
> The flakes seems to just be the same as those seen in master, not clear whether this PR makes it worse.

@halseth are they? I'm seeing blanket failures across _all_ the itests for this PR.

**Comment by halseth** [comment]
> > The flakes seems to just be the same as those seen in master, not clear whether this PR makes it worse.
> 
> @halseth are they? I'm seeing blanket failures across _all_ the itests for this PR.

Yeah, looks like this makes it worse indeed. I still think the root cause is fixed by #4953 though, probably some timing interaction of this PR that makes it more likely to flake.

**Comment by cfromknecht** [comment]
@halseth @Roasbeef let's merge #5003 and rebase over that to see if those flakes go away. if so i think this is ready to land

**Comment by halseth** [comment]
Rebased on #5003 to check travis

**Comment by halseth** [comment]
rebased on master

**Review by cfromknecht — APPROVED** [review]
LGMTv2

File: discovery/gossiper.go
Comment (Roasbeef) [inline discovery/gossiper.go:1495]: 👍

**Review by Roasbeef — APPROVED** [review]
LGTM 🧃

**Pull request merged**
